File: data/device.go
package data

import (
	"fmt"
)

type Device interface {
	GetTimeDifference() int64
	GetType() string
}

type device struct {
	Time int64
	Type string
}

func NewDevice(timeDifference int64, deviceType string) Device {
	d := &device{
		Time: timeDifference,
		Type: deviceType,
	}
	fmt.Println("Neues Gerät erstellt: Time =", d.Time, ", Type =", d.Type)
	return d
}

func (d *device) GetTimeDifference() int64 {
	return d.Time
}

func (d *device) GetType() string {
	return d.Type
}

File: data/device.py
class Device:

    def __init__ (self, time, type):
        self.time = time
        self.type = type
        print(f"Created new device with time {time} and type {type}")

    def get_time(self):
        return self.time

    def get_type(self):
        return self.type
